packages/mux-player/src/video-api.ts

```diff
@@ -0,0 +1,243 @@
+import type MuxVideoElement from "@mux-elements/mux-video";
+
+const AllowedVideoAttributeNames = [
+  "autoplay",
+  "crossorigin",
+  "loop",
+  "muted",
+  "playsinline",
+  "src",
+  "poster",
+];
+
+const AllowedVideoEvents = [
+  "loadstart",
+  "loadedmetadata",
+  "progress",
+  "durationchange",
+  "volumechange",
+  "ratechange",
+  "resize",
+  "waiting",
+  "play",
+  "playing",
+  "timeupdate",
+  "pause",
+  "seeking",
+  "seeked",
+  "ended",
+];
+
+class VideoApiElement extends HTMLElement {
```

Comment:
Like we talked about since it's the base class for `MuxPlayerElement` and everything on this API interface is exposed to the public, best to have this class be an opt-in for HTMLVideoElement properties and methods. Slowly adding methods and properties when they are needed.

Reply:
yup, agreed and we discussed this sync 👍🏼